### ISSUE TITLE
Launcher pod “Terminating” immediately when be created

### ISSUE THREAD

**Issue by Longchuanzheng** [kubevirt/kubevirt#10123]
**What happened**:
When I create a vm with kubevirt 1.0, the launcher pod "Terminating" when be created. Meanwhile the annotations "kubevirt.io/keep-launcher-alive-after-failure: 'true'" doesn't work.
While I change to kubevirt 0.59, the pod can running and vm start success.

**What you expected to happen**:
Launcher pod running and vm start success.

**How to reproduce it (as minimally and precisely as possible)**:
Create vm like what I do with kubevirt 0.59

**Additional context**:
the vm yaml I exec :
```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: zal-vm-1719
spec:
  running: true
  template:
    metadata:
      labels:
        kubevirt.io/vm: zal-vm-1719
        kubevirt.io/domain: zal-vm-1719
      annotations:
        kubevirt.io/keep-launcher-alive-after-failure: 'true'
    spec:
      domain:
        machine:
          type: q35
        cpu:
          cores: 2
        resources:
          requests:
            memory: 1024M
        devices:
          autoattachMemBalloon: false
          disks:
          - disk:
              bus: virtio
            name: root-device
          - disk:
              bus: virtio
            name: cloudinitdisk
          interfaces:
          - name: default
            masquerade: {}
      terminationGracePeriodSeconds: 0
      networks:
      - name: default
        pod: {}
      volumes:
      - name: root-device
        persistentVolumeClaim:
          claimName: zal-centos-1
      - cloudInitConfigDrive:
          userData: |-
            #cloud-config
            disable_root: False
            password: root
            chpasswd:
                 list: |
                        root:root
                 expire: False
            ssh_pwauth: True
        name: cloudinitdisk
```

**Environment**:
- KubeVirt version (use `virtctl version`): 1.0
- Kubernetes version (use `kubectl version`): v1.21.5
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): 4.19
- Install tools: N/A
- Others: N/A

**Comment by Longchuanzheng** [comment]
launcher pod log:
```
[root@euler-x86-74 case]# kubectl logs virt-launcher-zal-vm-1719-csvk5
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:86","timestamp":"2023-07-17T09:12:16.326755Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:89","timestamp":"2023-07-17T09:12:16.326817Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu:///system","pos":"libvirt.go:545","timestamp":"2023-07-17T09:12:16.351046Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/virtqemud-sock': No such file or directory')","pos":"libvirt.go:553","timestamp":"2023-07-17T09:12:16.351443Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 9.0.0, package: 3.el9 (builder@centos.org, 2023-02-01-12:58:56, )","subcomponent":"libvirt","thread":"46","timestamp":"2023-07-17T09:12:16.376000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: zal-vm-1719","subcomponent":"libvirt","thread":"46","timestamp":"2023-07-17T09:12:16.376000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Child process (dmidecode -q -t 0,1,2,3,4,11,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2725","subcomponent":"libvirt","thread":"46","timestamp":"2023-07-17T09:12:16.376000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:561","timestamp":"2023-07-17T09:12:16.853148Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:543","timestamp":"2023-07-17T09:12:16.855354Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:75","timestamp":"2023-07-17T09:12:16.855511Z"}
{"component":"virt-launcher","level":"info","msg":"Received signal terminated","pos":"virt-launcher.go:457","timestamp":"2023-07-17T09:12:19.091911Z"}
{"component":"virt-launcher","level":"info","msg":"stopping cmd server","pos":"server.go:611","timestamp":"2023-07-17T09:12:19.091975Z"}
{"component":"virt-launcher","level":"info","msg":"cmd server stopped","pos":"server.go:620","timestamp":"2023-07-17T09:12:19.092164Z"}
{"component":"virt-launcher","level":"info","msg":"Exiting...","pos":"virt-launcher.go:493","timestamp":"2023-07-17T09:12:19.092206Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 30 with status 9","pos":"virt-launcher-monitor.go:125","timestamp":"2023-07-17T09:12:19.092852Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 29 with status 9","pos":"virt-launcher-monitor.go:125","timestamp":"2023-07-17T09:12:19.093064Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 12 with status 0","pos":"virt-launcher-monitor.go:125","timestamp":"2023-07-17T09:12:19.093843Z"}
```

virt-controller log:
```
[root@euler-x86-74 case]# kubectl logs -n kubevirt virt-controller-669966f9d4-4k4nb |grep zal-vm-1719
{"component":"virt-controller","kind":"","level":"error","msg":"Updating api version annotations failed","name":"zal-vm-1719","namespace":"default","pos":"vm.go:281","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.485396Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.485445Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating api version annotations failed","name":"zal-vm-1719","namespace":"default","pos":"vm.go:281","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.504189Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.504225Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:783","timestamp":"2023-07-17T09:12:03.515850Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance zal-vm-1719","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1020","timestamp":"2023-07-17T09:12:03.529038Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.549699Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:03.549741Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:14.358244Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/zal-vm-1719 with phase Failed","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1308","timestamp":"2023-07-17T09:12:14.363552Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:14.396753Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:12:14.408654Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:12:14.408709Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:14.408781Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:14.414105Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:783","timestamp":"2023-07-17T09:12:34.416003Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance zal-vm-1719","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1020","timestamp":"2023-07-17T09:12:34.428739Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.443848Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.443899Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.459003Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.459056Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.478864Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:12:34.478920Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:40.623082Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/zal-vm-1719 with phase Failed","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1308","timestamp":"2023-07-17T09:12:40.629355Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:40.666016Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:12:40.678767Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:12:40.678808Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:40.678865Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:12:40.684135Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 30 more seconds before starting.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:778","timestamp":"2023-07-17T09:12:54.408232Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 30 more seconds before starting.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:778","timestamp":"2023-07-17T09:12:54.417570Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Starting VM due to runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:783","timestamp":"2023-07-17T09:13:24.408543Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Started VM by creating the new virtual machine instance zal-vm-1719","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1020","timestamp":"2023-07-17T09:13:24.423574Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:13:24.438395Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:13:24.438451Z"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:13:24.454163Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"Operation cannot be fulfilled on virtualmachines.kubevirt.io \"zal-vm-1719\": the object has been modified; please apply your changes to the latest version and try again","timestamp":"2023-07-17T09:13:24.454214Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:13:31.020777Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Dispatching delete event for vmi default/zal-vm-1719 with phase Failed","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1308","timestamp":"2023-07-17T09:13:31.027190Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:13:31.061531Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"error","msg":"Updating the VirtualMachine status failed.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:359","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:13:31.075281Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","level":"info","msg":"re-enqueuing VirtualMachine default/zal-vm-1719","pos":"vm.go:247","reason":"the server rejected our request due to an error in our request","timestamp":"2023-07-17T09:13:31.075334Z"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:13:31.075380Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Stopping VM with VMI in phase Failed and VM runStrategy: Always","name":"zal-vm-1719","namespace":"default","pos":"vm.go:759","timestamp":"2023-07-17T09:13:31.080701Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 92 more seconds before starting.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:778","timestamp":"2023-07-17T09:13:43.070922Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
{"component":"virt-controller","kind":"","level":"info","msg":"Delaying start of VM Starting VM with 'runStrategy: Always' due to start failure backoff. Waiting 92 more seconds before starting.","name":"zal-vm-1719","namespace":"default","pos":"vm.go:778","timestamp":"2023-07-17T09:13:43.081634Z","uid":"645aba6f-4994-4f4b-9764-4cd9f720d760"}
```

virt-handler log :
```
[root@euler-x86-74 case]# kubectl logs -n kubevirt virt-handler-7fshb|grep zal-vm-1719
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Scheduled | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:17.075576Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"migration is block migration because of cloudinitdisk volume","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2447","timestamp":"2023-07-17T09:12:17.093632Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:17.107750Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid c98a2f31-5fdf-4941-be7a-65e4c4525bf7","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:17.107800Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:17.107834Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:17.107881Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:17.107931Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:17.114065Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid c98a2f31-5fdf-4941-be7a-65e4c4525bf7","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:17.114109Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:17.114150Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:17.114190Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:17.114252Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:17.132118Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid c98a2f31-5fdf-4941-be7a-65e4c4525bf7","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:17.132159Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:17.132189Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:17.132224Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:17.132262Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:17.146930Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid c98a2f31-5fdf-4941-be7a-65e4c4525bf7","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:17.146971Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:17.147008Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:17.147044Z","uid":"c98a2f31-5fdf-4941-be7a-65e4c4525bf7"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:17.147092Z","uid":""}
{"component":"virt-handler","kind":"VirtualMachineInstance","level":"info","msg":"Performing final local cleanup for vmi with uid ","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:37.166505Z","uid":""}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:37.166538Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Scheduled | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:43.337672Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"migration is block migration because of cloudinitdisk volume","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2447","timestamp":"2023-07-17T09:12:43.356138Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:43.372641Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid 853fa1f2-a61c-4e9e-acf6-d5dac4778154","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:43.372687Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:43.372736Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:43.372785Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:43.372834Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:43.379375Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid 853fa1f2-a61c-4e9e-acf6-d5dac4778154","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:43.379420Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:43.379454Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:43.379493Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:43.379537Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:43.398767Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid 853fa1f2-a61c-4e9e-acf6-d5dac4778154","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:43.398809Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:43.398841Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:43.398879Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:43.398919Z","uid":""}
{"component":"virt-handler","kind":"","level":"info","msg":"VMI is in phase: Failed | Domain does not exist","name":"zal-vm-1719","namespace":"default","pos":"vm.go:1777","timestamp":"2023-07-17T09:12:43.416057Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Performing final local cleanup for vmi with uid 853fa1f2-a61c-4e9e-acf6-d5dac4778154","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:43.416103Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"No container disk mount entries found to unmount","name":"zal-vm-1719","namespace":"default","pos":"mount.go:349","timestamp":"2023-07-17T09:12:43.416135Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"","level":"info","msg":"Cleaning up remaining hotplug volumes","name":"zal-vm-1719","namespace":"default","pos":"mount.go:769","timestamp":"2023-07-17T09:12:43.416173Z","uid":"853fa1f2-a61c-4e9e-acf6-d5dac4778154"}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:43.416212Z","uid":""}
{"component":"virt-handler","kind":"VirtualMachineInstance","level":"info","msg":"Performing final local cleanup for vmi with uid ","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2041","timestamp":"2023-07-17T09:12:57.158017Z","uid":""}
{"component":"virt-handler","kind":"Domain","level":"info","msg":"Removing domain from cache during final cleanup","name":"zal-vm-1719","namespace":"default","pos":"vm.go:2075","timestamp":"2023-07-17T09:12:57.158053Z","uid":""}
```

**Comment by jiridanek** [comment]
I'm seeing the same thing. I am also seeing this in the launcher pod log

```
│ compute {"component":"virt-launcher","level":"error","msg":"internal error: Child process (dmidecode -q -t 0,1,2,3,4,11,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2725","subcomponent":"l │
│ ibvirt","thread":"34","timestamp":"2023-07-17T22:14:24.627000Z"}   
```

**Comment by Longchuanzheng** [comment]
> I'm seeing the same thing. I am also seeing this in the launcher pod log
> 
> ```
> │ compute {"component":"virt-launcher","level":"error","msg":"internal error: Child process (dmidecode -q -t 0,1,2,3,4,11,17) unexpected exit status 1: /dev/mem: No such file or directory","pos":"virCommandWait:2725","subcomponent":"l │
> │ ibvirt","thread":"34","timestamp":"2023-07-17T22:14:24.627000Z"}   
> ```

@jiridanek  I think this log just due to the components start order,  I find these log in a normal running launcher pod.

**Comment by jiridanek** [comment]
I was trying to exec the testing cirros VM,

```
$ kubectl apply -f https://kubevirt.io/labs/manifests/vm.yaml
$ ./virtctl-v1.0.0-linux-amd64 start testvm
$ ./virtctl-v1.0.0-linux-amd64 console testvm
```

The actual issues I encountered and resolved were

* I had a machine without `libvirtd` running
* Had to enable `Root` feature gate
   * https://kubevirt.io/user-guide/operations/activating_feature_gates/
   * https://github.com/kubevirt/kubevirt/issues/9784#issuecomment-1638963029
* I used k0s, so I had to deal with https://github.com/kubevirt/kubevirt/issues/8429#issuecomment-1241728252
   * The problem is that the initial workaround, consisting of symlinking `/var/lib/k0s/kubelet /var/lib/kubelet` stopped working, because (for security reasons? https://github.com/kubevirt/kubevirt/issues/5913#issuecomment-1376882097) links are no longer followed
   * The current solution is to use a bind mount, instead of symlink, as shown in https://github.com/kubevirt/kubevirt/issues/8387#issuecomment-1463472051
   * see also https://github.com/k0sproject/k0s/issues/428

**Comment by zakkg3** [comment]
> I was trying to exec the testing cirros VM,
> 
> ```
> $ kubectl apply -f https://kubevirt.io/labs/manifests/vm.yaml
> $ ./virtctl-v1.0.0-linux-amd64 start testvm
> $ ./virtctl-v1.0.0-linux-amd64 console testvm
> ```
> 
> The actual issues I encountered and resolved were
> 
> * I had a machine without `libvirtd` running
> * Had to enable `Root` feature gate
>   
>   * https://kubevirt.io/user-guide/operations/activating_feature_gates/
>   * [when i follow Labs - KubeVirt quickstart with Minikube,i can't strart vm #9784 (comment)](https://github.com/kubevirt/kubevirt/issues/9784#issuecomment-1638963029)
> * I used k0s, so I had to deal with [create vmi failed with official cirros containerdisk #8429 (comment)](https://github.com/kubevirt/kubevirt/issues/8429#issuecomment-1241728252)
>   
>   * The problem is that the initial workaround, consisting of symlinking `/var/lib/k0s/kubelet /var/lib/kubelet` stopped working, because (for security reasons? [Allow configuration of alternative kubelet directories #5913 (comment)](https://github.com/kubevirt/kubevirt/issues/5913#issuecomment-1376882097)) links are no longer followed
>   * The current solution is to use a bind mount, instead of symlink, as shown in [virt-chroot mount expects a directory #8387 (comment)](https://github.com/kubevirt/kubevirt/issues/8387#issuecomment-1463472051)
>   * see also [Symlink podVolumePath k0sproject/k0s#428](https://github.com/k0sproject/k0s/issues/428)
> * SELinux issues
>   
>   * `{"component":"virt-handler","level":"error","msg":"couldn't find the SELinux context for minibench-0","pos":"vm.go:1275","reason":"No command socket fou │ │ nd for vmi a5eb6822-a0c0-4197-ad9b-3f5bbd50e81e","timestamp":"2023-07-18T13:48:13.030300Z"}`
>   * `setenforce permissive` and bye!

same stack same problems.
thanks for sharing, helped to save some time.

note you dont need libvirtd running in your machine. kubevirt comes with it. if you need it for other reason, you have to update livbirt's apparmor with `/usr/libexec/qemu-kvm PUx,`

**Comment by gzimin** [comment]
> > I was trying to exec the testing cirros VM,
> > ```
> > $ kubectl apply -f https://kubevirt.io/labs/manifests/vm.yaml
> > $ ./virtctl-v1.0.0-linux-amd64 start testvm
> > $ ./virtctl-v1.0.0-linux-amd64 console testvm
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The actual issues I encountered and resolved were
> > 
> > * I had a machine without `libvirtd` running
> > * Had to enable `Root` feature gate
> >   
> >   * https://kubevirt.io/user-guide/operations/activating_feature_gates/
> >   * [when i follow Labs - KubeVirt quickstart with Minikube,i can't strart vm #9784 (comment)](https://github.com/kubevirt/kubevirt/issues/9784#issuecomment-1638963029)
> > * I used k0s, so I had to deal with [create vmi failed with official cirros containerdisk #8429 (comment)](https://github.com/kubevirt/kubevirt/issues/8429#issuecomment-1241728252)
> >   
> >   * The problem is that the initial workaround, consisting of symlinking `/var/lib/k0s/kubelet /var/lib/kubelet` stopped working, because (for security reasons? [Allow configuration of alternative kubelet directories #5913 (comment)](https://github.com/kubevirt/kubevirt/issues/5913#issuecomment-1376882097)) links are no longer followed
> >   * The current solution is to use a bind mount, instead of symlink, as shown in [virt-chroot mount expects a directory #8387 (comment)](https://github.com/kubevirt/kubevirt/issues/8387#issuecomment-1463472051)
> >   * see also [Symlink podVolumePath k0sproject/k0s#428](https://github.com/k0sproject/k0s/issues/428)
> > * SELinux issues
> >   
> >   * `{"component":"virt-handler","level":"error","msg":"couldn't find the SELinux context for minibench-0","pos":"vm.go:1275","reason":"No command socket fou │ │ nd for vmi a5eb6822-a0c0-4197-ad9b-3f5bbd50e81e","timestamp":"2023-07-18T13:48:13.030300Z"}`
> >   * `setenforce permissive` and bye!
> 
> almost same stack same problems. thanks for sharing, helped to save some time.
> 
> Note 1:
> 
> I have to add `--feature-gates=DevicePlugins=true` on the kube worker system unit. So its: `ExecStart=/usr/local/bin/k0s worker --data-dir=/var/lib/k0s --kubelet-extra-args=--node-ip=10.205.202.139,--feature-gates=DevicePlugins=true --token-file=/etc/k0s/k0stoken` see this [NVIDIA/k8s-device-plugin#10 (comment)](https://github.com/NVIDIA/k8s-device-plugin/issues/10#issuecomment-346846478)
> 
> note 2:
> 
> featureGates -roots its not needed for me.
> 
> note 3: you dont need libvirtd running in your machine. kubevirt comes with it. if you need it for other reason, you have to update livbirt's apparmor with `/usr/libexec/qemu-kvm PUx,`

Hello, got the same problem. I followed all advices here, but still can't boot vm properly.
After I mounted k0s kubelet dir to the default path, I restart k0sworker services, but then I got this error in virt-launcher pod
`Warning  UnexpectedAdmissionError  16s   kubelet            Allocate failed due to cannot allocate unregistered device devices.kubevirt.io/tun, which is unexpected` When I'm just restarting node virt-launcher pod starts okay, but it is not resolving the initial issue.
I tried to use kubevirt with k0s, version of k0s `1.27.5+k0s.0`
